app/presenters/oauth_metadata_presenter.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+class OauthMetadataPresenter < ActiveModelSerializers::Model
+  include RoutingHelper
+
+  attributes :issuer, :authorization_endpoint, :token_endpoint,
+             :registration_endpoint, :revocation_endpoint, :scopes_supported,
+             :response_types_supported, :response_modes_supported,
+             :grant_types_supported, :token_endpoint_auth_methods_supported
+
+  def issuer
+    root_url
+  end
+
+  def authorization_endpoint
+    oauth_authorization_url
+  end
+
+  def token_endpoint
+    oauth_token_url
+  end
+
+  # NOTE: the api_v1_apps route doesn't technically conform to the
+  # specification for OAuth 2.0 Dynamic Client Registration defined in
+  # https://datatracker.ietf.org/doc/html/rfc7591
+  # But for Mastodon, this is the API Endpoint that you would want for this property
+  def registration_endpoint
+    api_v1_apps_url
+  end
```

Comment:
On the one hand, I don't think we want to advertise with this property an endpoint that does *not* support the spec… but on the other hand, having this is some very important piece of information for clients…

Maybe we should start by not advertising this, but having a custom property instead, and add `service_documentation` to point to the Mastodon API documentation?

Reply:
Yeah, we could totally do that too. I do want to do a PR to properly support Dynamic Client Registration (which would then allow for us to support something like Solid's OIDC Public Identifier Document schema, where you register with a URL and the server fetches the document at that URL for the registration).

Reply:
Sounds like a plan to me!

Reply:
@ClearlyClaire have updated this PR based on prior conversations.